src/Execution/Executor.php

```diff
@@ -653,67 +589,81 @@ private function completeAbstractValue(
     }
 
     /**
-     * @param                       $runtimeTypeOrName
-     * @param AbstractTypeInterface $returnType
-     * @param array                 $fieldNodes
-     * @param ResolveInfo           $info
-     * @param                       $result
+     * @param NamedTypeInterface|string $runtimeTypeOrName
+     * @param AbstractTypeInterface     $returnType
+     * @param ResolveInfo               $info
+     * @param mixed                     $result
      * @return TypeInterface|ObjectType|null
      * @throws ExecutionException
-     * @throws \Digia\GraphQL\Error\InvariantException
+     * @throws InvariantException
      */
-    private function ensureValidRuntimeType(
+    protected function ensureValidRuntimeType(
         $runtimeTypeOrName,
         AbstractTypeInterface $returnType,
-        array $fieldNodes,
         ResolveInfo $info,
         &$result
     ) {
-        $runtimeType = is_string($runtimeTypeOrName)
+        /** @var NamedTypeInterface $runtimeType */
+        $runtimeType = \is_string($runtimeTypeOrName)
             ? $this->context->getSchema()->getType($runtimeTypeOrName)
             : $runtimeTypeOrName;
 
-        $runtimeTypeName = is_string($runtimeType) ?: $runtimeType->getName();
+        $runtimeTypeName = $runtimeType->getName();
         $returnTypeName  = $returnType->getName();
 
         if (!$runtimeType instanceof ObjectType) {
             $parentTypeName = $info->getParentType()->getName();
             $fieldName      = $info->getFieldName();
 
             throw new ExecutionException(
-                "Abstract type {$returnTypeName} must resolve to an Object type at runtime " .
-                "for field {$parentTypeName}.{$fieldName} with " .
-                'value "' . $result . '", received "{$runtimeTypeName}".'
+                \sprintf(
```

Comment:
I intentionally avoid using `sprintf` for long message because it's very hard to know which variable in which placeholder.

Reply:
Again, consistency and also readaility.

Reply:
I don't think it's easier to read :D If consistency open a door for potential issue I would I think again. Context is of usage is also important.

Reply:
A potential issue with an error message? I wouldn't sacrifice consistency for that.

Reply:
We have numerous similar `sprintf` statements in the `Schema` package so changing this back won't really solve the issue alone. I would change this for consistency now and open a proposal for not using `sprintf` for very long error messages.